From dcc9d3e58fd2c35d6648c42cd1498e670615669d Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Tue, 26 Jul 2022 10:03:58 +0800
Subject: [PATCH] sw64: remove context check in csum_partial_copy_from_user()
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: dcc9d3e58fd2c35d6648c42cd1498e670615669d
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5PNSZ

--------------------------------

This reverts commit ce6455155eeacae6557fe0142e69a6c288ac970e.

In previous patch, we fixed deep-copy_user(). It's now safe to use
it in any context, so no need to check.

Signed-off-by: Mao Minkai <maominkai@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/lib/csum_partial_copy.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/arch/sw_64/lib/csum_partial_copy.c b/arch/sw_64/lib/csum_partial_copy.c
index 742dd63cdb70..1a8c18757e09 100644
--- a/arch/sw_64/lib/csum_partial_copy.c
+++ b/arch/sw_64/lib/csum_partial_copy.c
@@ -61,10 +61,7 @@ csum_partial_cfu_dest_aligned(const unsigned long __user *src,
 	unsigned long checksum = ~0U;
 	int err = 0;
 
-	if (likely(!uaccess_kernel()))
-		err = __copy_from_user(dst, src, len + 8);
-	else
-		memcpy(dst, src, len + 8);
+	err = __copy_from_user(dst, src, len+8);
 
 	while (len > 0) {
 		word = *dst;
@@ -93,10 +90,7 @@ csum_partial_cfu_dest_unaligned(const unsigned long __user *src,
 	unsigned long checksum = ~0U;
 	int err = 0;
 
-	if (likely(!uaccess_kernel()))
-		err = __copy_from_user(dst, src, len + 8);
-	else
-		memcpy(dst, src, len + 8);
+	err = __copy_from_user(dst, src, len+8);
 
 	dst = (unsigned long *)((unsigned long)dst & (~7UL));
 	word = *dst;
-- 
2.33.0

